feature-libs/tracking/tms/tms.module.ts

```diff
@@ -0,0 +1,36 @@
+import { APP_INITIALIZER, ModuleWithProviders, NgModule } from '@angular/core';
+import { provideConfig, provideDefaultConfig } from '@spartacus/core';
+import { defaultAdobeExperiencePlatformConfig } from '@spartacus/tracking/tms/aep';
+import { TmsConfig, TmsService } from '@spartacus/tracking/tms/core';
+import { defaultGoogleTagManagerConfig } from '@spartacus/tracking/tms/gtm';
+
+/**
+ * The factory that conditionally (based on the configuration) starts collecting events
+ */
+export function tmsFactory(service: TmsService): () => void {
+  const result = () => service.collect();
+  return result;
+}
+
+@NgModule({
+  providers: [
+    {
```

Comment:
if you have forRoot, APP_INITIALIZER should be in the root providers. 
Also, all this should be part of TmsCoreModule IMO (+Gtm and Adobe specific modules) and this one could include those.